qpc/insights/tests_insights_configure.py

```diff
@@ -0,0 +1,87 @@
+"""Test the CLI module."""
+
+import sys
+import unittest
+from io import StringIO
+
+from qpc.cli import CLI
+from qpc.tests_utilities import HushUpStderr, redirect_stdout
+from qpc.utils import read_insights_config, write_insights_config, write_server_config
+
+DEFAULT_PORT = 443
+DEFAULT_HOST = 'console.redhat.com'
+
+DEFAULT_CONFIG = {'host': '127.0.0.1', 'port': 8000, 'use_http': True, 'require_token': False}
+
+
+class InsightsConfigureTests(unittest.TestCase):
+    """Class for testing insights host configuration."""
+
+    def setUp(self):
+        """Create test setup."""
+        write_server_config(DEFAULT_CONFIG)
+        # Temporarily disable stderr for these tests, CLI errors clutter up
+        self.orig_stderr = sys.stderr
+        sys.stderr = HushUpStderr()
+
+    def tearDown(self):
+        """Remove test case setup."""
+        # Reset insights config to default host/port
+        config_out = StringIO()
+        sys.argv = ['/bin/qpc', 'insights', 'config',
+                    '--host', str(DEFAULT_HOST), '--port', str(DEFAULT_PORT)]
+
+        with redirect_stdout(config_out):
+            CLI().main()
+            config = read_insights_config()
+            self.assertEqual(config['host'], DEFAULT_HOST)
+            self.assertEqual(config['port'], DEFAULT_PORT)
+
+        # Restore stderr
+        sys.stderr = self.orig_stderr
+
```

Comment:
Should the teardown also remove the config file that was created in the setup?

Reply:
The tmp file fixture that Bruno created will take care of that. It won't modify the real config file.